File: src/Samples/CSharpCodeSamples/QueueingServiceExamples.cs
namespace CSharpCodeSamples
{
    using System;
    using System.Collections.ObjectModel;
    using System.Net.Http;
    using System.Threading;
    using System.Threading.Tasks;
    using OpenStack.Collections;
    using OpenStack.ObjectModel.JsonHome;
    using OpenStack.Security.Authentication;
    using OpenStack.Services.Queues.V1;
    using Rackspace.Threading;

    public class QueueingServiceExamples
    {
        // use the default region for the account
        string region = null;
        // create a new client ID for this instance
        Guid clientId = Guid.NewGuid();
        // access Cloud Queues over the public Internet
        bool internalUrl = false;
        // use a default RackspaceAuthenticationClient for authentication
        IAuthenticationService authenticationService = null;

        #region GetHomeAsync
        public async Task PrepareGetHomeAsyncAwait()
        {
            #region PrepareGetHomeAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            GetHomeApiCall apiCall = await queuesService.PrepareGetHomeAsync(CancellationToken.None);
            Tuple<HttpResponseMessage, HomeDocument> apiResponse = await apiCall.SendAsync(CancellationToken.None);
            HomeDocument homeDocument = apiResponse.Item2;
            #endregion
        }

        public void PrepareGetHome()
        {
            #region PrepareGetHomeAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            Task<HomeDocument> homeDocumentTask =
                CoreTaskExtensions.Using(
                    () => queuesService.PrepareGetHomeAsync(CancellationToken.None),
                    task => task.Result.SendAsync(CancellationToken.None))
                .Select(task => task.Result.Item2);
            #endregion
        }

        public async Task GetHomeAsyncAwait()
        {
            #region GetHomeAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            HomeDocument createdQueue = await queuesService.GetHomeAsync(CancellationToken.None);
            #endregion
        }

        public void GetHome()
        {
            #region GetHomeAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            Task<HomeDocument> task = queuesService.GetHomeAsync(CancellationToken.None);
            #endregion
        }
        #endregion

        #region GetNodeHealthAsync
        public async Task PrepareGetNodeHealthAsyncAwait()
        {
            #region PrepareGetNodeHealthAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            GetNodeHealthApiCall apiCall = await queuesService.PrepareGetNodeHealthAsync(CancellationToken.None);
            Tuple<HttpResponseMessage, bool> apiResponse = await apiCall.SendAsync(CancellationToken.None);
            bool operational = apiResponse.Item2;
            #endregion
        }

        public void PrepareGetNodeHealth()
        {
            #region PrepareGetNodeHealthAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            Task<bool> nodeHealthTask =
                CoreTaskExtensions.Using(
                    () => queuesService.PrepareGetNodeHealthAsync(CancellationToken.None),
                    task => task.Result.SendAsync(CancellationToken.None))
                .Select(task => task.Result.Item2);
            #endregion
        }

        public async Task GetNodeHealthAsyncAwait()
        {
            #region GetNodeHealthAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            bool operational = await queuesService.GetNodeHealthAsync(CancellationToken.None);
            #endregion
        }

        public void GetNodeHealth()
        {
            #region GetNodeHealthAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            Task<bool> task = queuesService.GetNodeHealthAsync(CancellationToken.None);
            #endregion
        }
        #endregion

        #region CreateQueueAsync
        public async Task PrepareCreateQueueAsyncAwait()
        {
            #region PrepareCreateQueueAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            CreateQueueApiCall apiCall = await queuesService.PrepareCreateQueueAsync(queueName, CancellationToken.None);
            Tuple<HttpResponseMessage, bool> apiResponse = await apiCall.SendAsync(CancellationToken.None);
            bool createdQueue = apiResponse.Item2;
            #endregion
        }

        public void PrepareCreateQueue()
        {
            #region PrepareCreateQueueAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task<bool> createQueueTask =
                CoreTaskExtensions.Using(
                    () => queuesService.PrepareCreateQueueAsync(queueName, CancellationToken.None),
                    task => task.Result.SendAsync(CancellationToken.None))
                .Select(task => task.Result.Item2);
            #endregion
        }

        public async Task CreateQueueAsyncAwait()
        {
            #region CreateQueueAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            bool createdQueue = await queuesService.CreateQueueAsync(queueName, CancellationToken.None);
            #endregion
        }

        public void CreateQueue()
        {
            #region CreateQueueAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task<bool> task = queuesService.CreateQueueAsync(queueName, CancellationToken.None);
            #endregion
        }
        #endregion

        #region RemoveQueueAsync
        public async Task PrepareRemoveQueueAsyncAwait()
        {
            #region PrepareRemoveQueueAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            RemoveQueueApiCall apiCall = await queuesService.PrepareRemoveQueueAsync(queueName, CancellationToken.None);
            Tuple<HttpResponseMessage, string> apiResponse = await apiCall.SendAsync(CancellationToken.None);
            #endregion
        }

        public void PrepareRemoveQueue()
        {
            #region PrepareRemoveQueueAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task removeQueueTask =
                CoreTaskExtensions.Using(
                    () => queuesService.PrepareRemoveQueueAsync(queueName, CancellationToken.None),
                    task => task.Result.SendAsync(CancellationToken.None));
            #endregion
        }

        public async Task RemoveQueueAsyncAwait()
        {
            #region RemoveQueueAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            await queuesService.RemoveQueueAsync(queueName, CancellationToken.None);
            #endregion
        }

        public void RemoveQueue()
        {
            #region RemoveQueueAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task task = queuesService.RemoveQueueAsync(queueName, CancellationToken.None);
            #endregion
        }
        #endregion

        #region ListQueuesAsync
        public async Task PrepareListQueuesAsyncAwait()
        {
            #region PrepareListQueuesAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            ListQueuesApiCall apiCall = await queuesService.PrepareListQueuesAsync(CancellationToken.None).WithDetails();
            Tuple<HttpResponseMessage, ReadOnlyCollectionPage<Queue>> apiResponse = await apiCall.SendAsync(CancellationToken.None);
            ReadOnlyCollectionPage<Queue> firstPage = apiResponse.Item2;
            ReadOnlyCollection<Queue> queues = await firstPage.GetAllPagesAsync(CancellationToken.None, null);
            #endregion
        }

        public void PrepareListQueues()
        {
            #region PrepareListQueuesAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            Task<ReadOnlyCollectionPage<Queue>> listQueuesTask =
                CoreTaskExtensions.Using(
                    () => queuesService.PrepareListQueuesAsync(CancellationToken.None).WithDetails(),
                    task => task.Result.SendAsync(CancellationToken.None))
                .Select(task => task.Result.Item2);
            Task<ReadOnlyCollection<Queue>> allQueuesTask =
                listQueuesTask.Then(task => task.Result.GetAllPagesAsync(CancellationToken.None, null));
            #endregion
        }

        public async Task ListQueuesAsyncAwait()
        {
            #region ListQueuesAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            ReadOnlyCollectionPage<Queue> queuesPage = await queuesService.ListQueuesAsync(true, CancellationToken.None);
            ReadOnlyCollection<Queue> queues = await queuesPage.GetAllPagesAsync(CancellationToken.None, null);
            #endregion
        }

        public void ListQueues()
        {
            #region ListQueuesAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            Task<ReadOnlyCollectionPage<Queue>> queuesPageTask = queuesService.ListQueuesAsync(true, CancellationToken.None);
            Task<ReadOnlyCollection<Queue>> queuesTask =
                queuesPageTask.Then(task => task.Result.GetAllPagesAsync(CancellationToken.None, null));
            #endregion
        }
        #endregion

        #region QueueExistsAsync
        public async Task PrepareQueueExistsAsyncAwait()
        {
            #region PrepareQueueExistsAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            QueueExistsApiCall apiCall = await queuesService.PrepareQueueExistsAsync(queueName, CancellationToken.None);
            Tuple<HttpResponseMessage, bool> apiResponse = await apiCall.SendAsync(CancellationToken.None);
            bool queueExists = apiResponse.Item2;
            #endregion
        }

        public void PrepareQueueExists()
        {
            #region PrepareQueueExistsAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task<bool> queueExistsTask =
                CoreTaskExtensions.Using(
                    () => queuesService.PrepareQueueExistsAsync(queueName, CancellationToken.None),
                    task => task.Result.SendAsync(CancellationToken.None))
                .Select(task => task.Result.Item2);
            #endregion
        }

        public async Task QueueExistsAsyncAwait()
        {
            #region QueueExistsAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            bool exists = await queuesService.QueueExistsAsync(queueName, CancellationToken.None);
            #endregion
        }

        public void QueueExists()
        {
            #region QueueExistsAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task<bool> task = queuesService.QueueExistsAsync(queueName, CancellationToken.None);
            #endregion
        }
        #endregion

        #region ListMessagesAsync
        public async Task PrepareListMessagesAsyncAwait()
        {
            #region PrepareListMessagesAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            ListMessagesApiCall apiCall = await queuesService.PrepareListMessagesAsync(queueName, CancellationToken.None);
            Tuple<HttpResponseMessage, ReadOnlyCollectionPage<QueuedMessage>> apiResponse = await apiCall.SendAsync(CancellationToken.None);
            ReadOnlyCollectionPage<QueuedMessage> firstPage = apiResponse.Item2;
            ReadOnlyCollection<QueuedMessage> messages = await firstPage.GetAllPagesAsync(CancellationToken.None, null);
            #endregion
        }

        public void PrepareListMessages()
        {
            #region PrepareListMessagesAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task<ReadOnlyCollectionPage<QueuedMessage>> listMessagesTask =
                CoreTaskExtensions.Using(
                    () => queuesService.PrepareListMessagesAsync(queueName, CancellationToken.None),
                    task => task.Result.SendAsync(CancellationToken.None))
                .Select(task => task.Result.Item2);
            Task<ReadOnlyCollection<QueuedMessage>> allMessagesTask =
                listMessagesTask.Then(task => task.Result.GetAllPagesAsync(CancellationToken.None, null));
            #endregion
        }

        public async Task ListMessagesAsyncAwait()
        {
            #region ListMessagesAsync (await)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            ReadOnlyCollectionPage<QueuedMessage> messagesPage = await queuesService.ListMessagesAsync(queueName, CancellationToken.None);
            ReadOnlyCollection<QueuedMessage> messages = await messagesPage.GetAllPagesAsync(CancellationToken.None, null);
            #endregion
        }

        public void ListMessages()
        {
            #region ListMessagesAsync (TPL)
            IQueuesService queuesService = new QueuesClient(authenticationService, region, clientId, internalUrl);
            QueueName queueName = new QueueName("ExampleQueue");
            Task<ReadOnlyCollectionPage<QueuedMessage>> messagesPageTask = queuesService.ListMessagesAsync(queueName, CancellationToken.None);
            Task<ReadOnlyCollection<QueuedMessage>> messagesTask =
                messagesPageTask.Then(task => task.Result.GetAllPagesAsync(CancellationToken.None, null));
            #endregion
        }
        #endregion
    }
}

File: src/Samples/VBCodeSamples/QueueingServiceExamples.vb
Imports System.Collections.ObjectModel
Imports System.Net.Http
Imports System.Threading
Imports System.Threading.Tasks
Imports OpenStack.Collections
Imports OpenStack.ObjectModel.JsonHome
Imports OpenStack.Security.Authentication
Imports OpenStack.Services.Queues.V1
Imports Rackspace.Threading

Public Class QueueingServiceExamples

    Dim region As String = Nothing
    Dim clientId = Guid.NewGuid
    Dim internalUrl = False
    Dim authenticationService As IAuthenticationService = Nothing

#Region "GetHomeAsync"
    Public Async Function PrepareGetHomeAsyncAwait() As Task
        ' #Region "PrepareGetHomeAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim apiCall As GetHomeApiCall = Await queuesService.PrepareGetHomeAsync(CancellationToken.None)
        Dim apiResponse As Tuple(Of HttpResponseMessage, HomeDocument) = Await apiCall.SendAsync(CancellationToken.None)
        Dim homeDocument As HomeDocument = apiResponse.Item2
        ' #End Region
    End Function

    Public Sub PrepareGetHome()
        ' #Region "PrepareGetHomeAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim homeDocumentTask As Task(Of HomeDocument) =
            CoreTaskExtensions.Using(
                Function() queuesService.PrepareGetHomeAsync(CancellationToken.None),
                Function(task) task.Result.SendAsync(CancellationToken.None)) _
            .Select(Function(task) task.Result.Item2)
        ' #End Region
    End Sub

    Public Async Function GetHomeAsyncAwait() As Task
        ' #Region "GetHomeAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim homeDocument = Await queuesService.GetHomeAsync(CancellationToken.None)
        ' #End Region
    End Function

    Public Sub GetHome()
        ' #Region "GetHomeAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim task = queuesService.GetHomeAsync(CancellationToken.None)
        ' #End Region
    End Sub
#End Region

#Region "GetNodeHealthAsync"
    Public Async Function PrepareGetNodeHealthAsyncAwait() As Task
        ' #Region "PrepareGetNodeHealthAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim apiCall As GetNodeHealthApiCall = Await queuesService.PrepareGetNodeHealthAsync(CancellationToken.None)
        Dim apiResponse As Tuple(Of HttpResponseMessage, Boolean) = Await apiCall.SendAsync(CancellationToken.None)
        Dim operational As Boolean = apiResponse.Item2
        ' #End Region
    End Function

    Public Sub PrepareGetNodeHealth()
        ' #Region "PrepareGetNodeHealthAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim nodeHealthTask As Task(Of Boolean) =
            CoreTaskExtensions.Using(
                Function() queuesService.PrepareGetNodeHealthAsync(CancellationToken.None),
                Function(task) task.Result.SendAsync(CancellationToken.None)) _
            .Select(Function(task) task.Result.Item2)
        ' #End Region
    End Sub

    Public Async Function GetNodeHealthAsyncAwait() As Task
        ' #Region "GetNodeHealthAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim operational As Boolean = Await queuesService.GetNodeHealthAsync(CancellationToken.None)
        ' #End Region
    End Function

    Public Sub GetNodeHealth()
        ' #Region "GetNodeHealthAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim task As Task(Of Boolean) = queuesService.GetNodeHealthAsync(CancellationToken.None)
        ' #End Region
    End Sub
#End Region

#Region "CreateQueueAsync"
    Public Async Function PrepareCreateQueueAsyncAwait() As Task
        ' #Region "PrepareCreateQueueAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim apiCall As CreateQueueApiCall = Await queuesService.PrepareCreateQueueAsync(queueName, CancellationToken.None)
        Dim apiResponse As Tuple(Of HttpResponseMessage, Boolean) = Await apiCall.SendAsync(CancellationToken.None)
        Dim createdQueue As Boolean = apiResponse.Item2
        ' #End Region
    End Function

    Public Sub PrepareCreateQueue()
        ' #Region "PrepareCreateQueueAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim queueCreatedTask As Task(Of Boolean) =
            CoreTaskExtensions.Using(
                Function() queuesService.PrepareCreateQueueAsync(queueName, CancellationToken.None),
                Function(task) task.Result.SendAsync(CancellationToken.None)) _
            .Select(Function(task) task.Result.Item2)
        ' #End Region
    End Sub

    Public Async Function CreateQueueAsyncAwait() As Task
        ' #Region "CreateQueueAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim createdQueue = Await queuesService.CreateQueueAsync(queueName, CancellationToken.None)
        ' #End Region
    End Function

    Public Sub CreateQueue()
        ' #Region "CreateQueueAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim task = queuesService.CreateQueueAsync(queueName, CancellationToken.None)
        ' #End Region
    End Sub
#End Region

#Region "RemoveQueueAsync"
    Public Async Function PrepareRemoveQueueAsyncAwait() As Task
        ' #Region "PrepareRemoveQueueAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim apiCall As RemoveQueueApiCall = Await queuesService.PrepareRemoveQueueAsync(queueName, CancellationToken.None)
        Dim apiResponse As Tuple(Of HttpResponseMessage, String) = Await apiCall.SendAsync(CancellationToken.None)
        ' #End Region
    End Function

    Public Sub PrepareRemoveQueue()
        ' #Region "PrepareRemoveQueueAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim queueCreatedTask As Task =
            CoreTaskExtensions.Using(
                Function() queuesService.PrepareRemoveQueueAsync(queueName, CancellationToken.None),
                Function(task) task.Result.SendAsync(CancellationToken.None))
        ' #End Region
    End Sub

    Public Async Function RemoveQueueAsyncAwait() As Task
        ' #Region "RemoveQueueAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Await queuesService.RemoveQueueAsync(queueName, CancellationToken.None)
        ' #End Region
    End Function

    Public Sub RemoveQueue()
        ' #Region "RemoveQueueAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim task = queuesService.RemoveQueueAsync(queueName, CancellationToken.None)
        ' #End Region
    End Sub
#End Region

#Region "ListQueuesAsync"
    Public Async Function PrepareListQueuesAsyncAwait() As Task
        ' #Region "PrepareListQueuesAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim apiCall As ListQueuesApiCall = Await queuesService.PrepareListQueuesAsync(CancellationToken.None).WithDetails()
        Dim apiResponse As Tuple(Of HttpResponseMessage, ReadOnlyCollectionPage(Of Queue)) = Await apiCall.SendAsync(CancellationToken.None)
        Dim firstPage As ReadOnlyCollectionPage(Of Queue) = apiResponse.Item2
        Dim queues As ReadOnlyCollection(Of Queue) = Await firstPage.GetAllPagesAsync(CancellationToken.None, Nothing)
        ' #End Region
    End Function

    Public Sub PrepareListQueues()
        ' #Region "PrepareListQueuesAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim listQueuesTask As Task(Of ReadOnlyCollectionPage(Of Queue)) =
            CoreTaskExtensions.Using(
                Function() queuesService.PrepareListQueuesAsync(CancellationToken.None).WithDetails(),
                Function(task) task.Result.SendAsync(CancellationToken.None)) _
            .Select(Function(task) task.Result.Item2)
        Dim allQueuesTask As Task(Of ReadOnlyCollection(Of Queue)) =
            listQueuesTask.Then(Function(task) task.Result.GetAllPagesAsync(CancellationToken.None, Nothing))
        ' #End Region
    End Sub

    Public Async Function ListQueuesAsyncAwait() As Task
        ' #Region "ListQueuesAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queuesPage = Await queuesService.ListQueuesAsync(True, CancellationToken.None)
        Dim queues = Await queuesPage.GetAllPagesAsync(CancellationToken.None, Nothing)
        ' #End Region
    End Function

    Public Sub ListQueues()
        ' #Region "ListQueuesAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queuesPageTask = queuesService.ListQueuesAsync(True, CancellationToken.None)
        Dim queuesTask = queuesPageTask.Then(Function(task) task.Result.GetAllPagesAsync(CancellationToken.None, Nothing))
        ' #End Region
    End Sub
#End Region

#Region "QueueExistsAsync"
    Public Async Function PrepareQueueExistsAsyncAwait() As Task
        ' #Region "PrepareQueueExistsAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim apiCall As QueueExistsApiCall = Await queuesService.PrepareQueueExistsAsync(queueName, CancellationToken.None)
        Dim apiResponse As Tuple(Of HttpResponseMessage, Boolean) = Await apiCall.SendAsync(CancellationToken.None)
        Dim queueExists As Boolean = apiResponse.Item2
        ' #End Region
    End Function

    Public Sub PrepareQueueExists()
        ' #Region "PrepareQueueExistsAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim queueExistsTask As Task(Of Boolean) =
            CoreTaskExtensions.Using(
                Function() queuesService.PrepareQueueExistsAsync(queueName, CancellationToken.None),
                Function(task) task.Result.SendAsync(CancellationToken.None)) _
            .Select(Function(task) task.Result.Item2)
        ' #End Region
    End Sub

    Public Async Function QueueExistsAsyncAwait() As Task
        ' #Region "QueueExistsAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim exists = Await queuesService.QueueExistsAsync(queueName, CancellationToken.None)
        ' #End Region
    End Function

    Public Sub QueueExists()
        ' #Region "QueueExistsAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim task = queuesService.QueueExistsAsync(queueName, CancellationToken.None)
        ' #End Region
    End Sub
#End Region

#Region "ListMessagesAsync"
    Public Async Function PrepareListMessagesAsyncAwait() As Task
        ' #Region "PrepareListMessagesAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim apiCall As ListMessagesApiCall = Await queuesService.PrepareListMessagesAsync(queueName, CancellationToken.None)
        Dim apiResponse As Tuple(Of HttpResponseMessage, ReadOnlyCollectionPage(Of QueuedMessage)) = Await apiCall.SendAsync(CancellationToken.None)
        Dim firstPage As ReadOnlyCollectionPage(Of QueuedMessage) = apiResponse.Item2
        Dim messages As ReadOnlyCollection(Of QueuedMessage) = Await firstPage.GetAllPagesAsync(CancellationToken.None, Nothing)
        ' #End Region
    End Function

    Public Sub PrepareListMessages()
        ' #Region "PrepareListMessagesAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim listMessagesTask As Task(Of ReadOnlyCollectionPage(Of QueuedMessage)) =
            CoreTaskExtensions.Using(
                Function() queuesService.PrepareListMessagesAsync(queueName, CancellationToken.None),
                Function(task) task.Result.SendAsync(CancellationToken.None)) _
            .Select(Function(task) task.Result.Item2)
        Dim allMessagesTask As Task(Of ReadOnlyCollection(Of QueuedMessage)) =
            listMessagesTask.Then(Function(task) task.Result.GetAllPagesAsync(CancellationToken.None, Nothing))
        ' #End Region
    End Sub

    Public Async Function ListMessagesAsyncAwait() As Task
        ' #Region "ListMessagesAsync (await)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim messagesPage = Await queuesService.ListMessagesAsync(queueName, CancellationToken.None)
        Dim messages = Await messagesPage.GetAllPagesAsync(CancellationToken.None, Nothing)
        ' #End Region
    End Function

    Public Sub ListMessages()
        ' #Region "ListMessagesAsync (TPL)"
        Dim queuesService As IQueuesService = New QueuesClient(authenticationService, region, clientId, internalUrl)
        Dim queueName = New QueueName("ExampleQueue")
        Dim messagesPageTask = queuesService.ListMessagesAsync(queueName, CancellationToken.None)
        Dim messagesTask = messagesPageTask.Then(Function(task) task.Result.GetAllPagesAsync(CancellationToken.None, Nothing))
        ' #End Region
    End Sub
#End Region

End Class
